File: CSE3063F22_iteration2_classes/Registration.java
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;
import java.lang.reflect.Array;
import java.text.ParseException;
import java.util.ArrayList;
import java.util.List;

import org.json.simple.JSONArray;
import org.json.simple.JSONObject;
import org.json.simple.parser.JSONParser;

public class Registration {
    
    private Student student;
    private Course courses;
    private boolean isRegistered = false;
    
    public Registration(Student student, Course courses) {
        this.student = student;
        this.courses = courses;
    }
    
    public Student getStudent() {
        return student;
    }
    
    public Course getCourses() {
        return courses;
    }


    public boolean isProvidePrereqs(){
        boolean isIt=false;
        List<String> course_prereqs = getCourses().getPrerequisites();
        List<Course> student_transcript = getStudent().getTranscript().getpassedCourses();
        
 

        if(course_prereqs.isEmpty()){
          isIt=true;
         
        }

        else{
        
        int i=0;
        while(i<course_prereqs.size()){

        if((student_transcript.contains(course_prereqs.get(i)))){
            isIt=true;
       }
        i++;
        
    }
}
        return isIt;
    }

   
    public boolean isSemesterCorrect(){
       
            boolean isIt=false;
            int std_semester = Integer.valueOf(getStudent().getSemester());
    
            System.out.println("coure_semester: "+getCourses().getSemester());
    
            if(getCourses().getSemester().equals("T") || getCourses().getSemester().equals("E")){
                if(std_semester>6){
                    isIt=true;
                }
               }
    
            else if(getCourses().getSemester().equals("N")){
                if(std_semester>1){
                    isIt=true;
                }
               }
            
           else if(getCourses().getSemester().equals("U")){
                if(std_semester>6){
                    isIt=true;
                }
               }
            
            else if(getCourses().getSemester().equals("F")){
                if(std_semester>7){
                    isIt= true;
                }
               }
            else{
                int course_semester = Integer.parseInt(getCourses().getSemester());
                if(std_semester >= course_semester){
                    isIt= true;
                }
                else{
                    isIt= false;
                }
            }
    
            return isIt;
        }
    



}

File: ITERATION3/OOSD_3RD_ITERATİON_code/Registration.py
from Course import Course
from Student import Student


class Registration:
    def __init__(self, student: 'Student', courses: 'Course'):
        self.__student = student
        self.__courses = courses
    
    def getStudent(self) -> 'Student':
        return self.__student
    
    def set_student(self, student: 'Student'):
        self.__student = student
    
    def getCourses(self) -> 'Course':
        return self.__courses
    
    def set_courses(self, courses: 'Course'):
        self.__courses = courses
    
    def add_course(self, course: 'Course'):
        pass
    
    def remove_course(self, course: 'Course'):
        pass
    
    def is_provide_prereqs(self) -> bool:
        is_it = False
        course_prereqs = self.getCourses().get_prerequisites()
        student_transcript = self.getStudent().get_transcript().get_passed_courses()
        
        if not course_prereqs:
            is_it = True
        else:
            i = 0
            while i < len(course_prereqs):
                if course_prereqs[i] in student_transcript:
                    is_it = True
                i += 1
        return is_it
    
    def is_semester_correct(self):
        isIt=False
        std_semester = int(self.getStudent().getSemester())

        if self.getCourses().getSemester() in ("T", "E"):
            if std_semester > 6:
                isIt = True
        elif self.getCourses().getSemester() == "N":
            if std_semester > 1:
                isIt = True
        elif self.getCourses().getSemester() == "U":
            if std_semester > 6:
                isIt = True
        elif self.getCourses().getSemester() == "F":
            if std_semester > 7:
                isIt = True
        else:
            course_semester = int(self.getCourses().getSemester())
            if std_semester >= course_semester:
                isIt = True
            else:
                isIt = False

        return isIt